### ISSUE TITLE
Feature request: More specific mux control

### ISSUE THREAD

**Issue by Mahdi-zarei** [xchacha20-poly1305/husi#98]
Hello and thank you for your great project! I have started refactoring and maintaining the Nekoray project and found the ability to define mux setting for each outbound useful. I patched Nekoray to allow importing and exporting the mux setting using the exported links such that if mux == true then enable mux for that outbound and if == false disable it, and if not provided or it is any other value use default. I would appreciate your opinion on this and if you approve it I can also create a PR to add this functionality to Husi.
Thank you in advance

**Comment by xchacha20-poly1305** [comment]
Welcome to pull request! Set mux for every profile is useful. But I don't think it is a good idea to set unofficial link.

**Comment by Mahdi-zarei** [comment]
> Welcome to pull request! Set mux for every profile is useful. But I don't think it is a good idea to set unofficial link.

I agree that changing official procedures is wrong, but I have recently noticed that some new features are being added to proxies and there are no ways to export or import them using links (UoT or Mux for example) and also there is a catch that extra features are added to links via query parameters so in almost any sane implementation of link parsing, custom features (mux here) will be ignored so older clients and other clients that do not support mux import/export do as well support these links and simply ignore the extra parameters.

**Comment by xchacha20-poly1305** [comment]
@Mahdi-zarei You are right. But temporary, self-use changes may be adapted by others, leading to a terrible situation. See what happened to vmess link created by v2rayN we can know.
